src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+
+    public MyLinkedList() {
+        this.size = 0;
+        this.firstNode = null;
+        this.lastNode = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, lastNode, null);
+        if (this.isEmpty()) {
+            this.firstNode = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        this.lastNode = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) { // add to end of list or to empty list (index == size == 0)
+            this.add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(value, null, firstNode);
+            this.firstNode.prev = newNode;
+            this.firstNode = newNode;
+            size++;
+        } else {
+            Node<T> nextNode = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(value, nextNode.prev, nextNode);
+            nextNode.prev = newNode;
+            newNode.prev.next = newNode;
+            size++;
+        }
```

Comment:
```suggestion
        Node<T> nextNode = findNodeByIndex(index);
        Node<T> prevNode = nextNode .prev;
        Node<T> newNode = new Node<>(prevNode, value, nextNode);
        nextNode.prev = newNode;
        if (prevNode == null) {
            head = newNode;
        } else {
            prevNode.next = newNode;
        }
        size++;
```
Your solution is good but lets refactor this code a little bit
adding to the middle or to the beginning of the list differs only in one line so we can combine them